src/librustc_mir/build/scope.rs

```diff
@@ -267,29 +288,94 @@ impl<'a,'tcx> Builder<'a,'tcx> {
     pub fn extent_of_outermost_scope(&self) -> CodeExtent {
         self.scopes.first().map(|scope| scope.extent).unwrap()
     }
-}
 
-fn diverge_cleanup_helper<'tcx>(cfg: &mut CFG<'tcx>, scopes: &mut [Scope<'tcx>]) -> BasicBlock {
-    let len = scopes.len();
-
-    if len == 0 {
-        return DIVERGE_BLOCK;
+    pub fn panic_bound_check(&mut self,
```

Comment:
Cannot be tested for correctness at the moment because https://github.com/rust-lang/rust/issues/30474